src/os/windows/crashlog_win.cpp

```diff
@@ -55,21 +72,65 @@ class CrashLogWindows : public CrashLog {
 		return succeeded;
 	}
 
-	int WriteCrashDump() override
+	bool WriteCrashDump() override
 	{
-		return google_breakpad::ExceptionHandler::WriteMinidump(OTTD2FS(_personal_dir), MinidumpCallback, this) ? 1 : -1;
+		return google_breakpad::ExceptionHandler::WriteMinidump(OTTD2FS(_personal_dir), MinidumpCallback, this);
 	}
 #endif
 
+#if defined(_MSC_VER)
+	/* virtual */ bool TryExecute(std::string_view section_name, std::function<bool()> &&func) override
+	{
+		this->try_execute_active = true;
+		bool res;
+
+		__try {
+			res = func();
+		} __except (EXCEPTION_EXECUTE_HANDLER) {
+			fmt::print("Something went wrong when attempting to fill {} section of the crash log.\n", section_name);
```

Comment:
Printing won't show much in typical release run.

Reply:
True; but it also isn't all that important. The handler of the 'return false' will make it more visible. But for debugging at least it was useful.

Better suggestions? Or just leave it as it is?

Reply:
Handler of the 'return false' is also and fmt::print it seems.

Reply:
Somewhat. And there is still a bit of work to do here. But basically, it should reset the filename, so it doesn't show up in the error report. That is a bit the point of this `TryExcept`. But the current handling of all this is a bit awkward, but also not for this PR, I think :)

Reply:
Yeah this PR doesn't make it worse

Reply:
(mainly, MacOS currently always prints the filename, even if it is still empty .. so much stuff to fix here :D)